### ISSUE TITLE
[BUG] this.destroy is not a function

### ISSUE THREAD

**Issue by foxmicha** [C2FO/fast-csv#397]
**Describe the bug**

When an improperly formatted CSV is passed to the parser, the parser will emit an error. Destroying the stream when this error is received will result in the reference passed to the parser to no longer be defined.

**Parsing or Formatting?**

 - [ ] Formatting
 - [x] Parsing 

**To Reproduce**
1. Example file contents if applicable
This file contents, with headers on top, can be used to trigger the bug.
```
"Good", "OK", "BAD", "Empty"
"Good","Good","Good",""
"OK","OK","OK",""
"Bad","Bad","Bad"",""
"Broken","Broken","Broken",""
```

Line 4 contains the bad data.

2. Example code to reproduce the issue.
Some example / pseudo-ish code to reproduce this is below.
```js
var csv = require('fast-csv'),
  fs = require('fs'),
  path = require('path');

var stream = fs.createReadStream(path.resolve("testfile.csv"));

var csvStream = stream.pipe(csv.parse({
     ignoreEmpty: true
}));

csvStream.on("data", function(r) {
    console.log(r);
});

csvStream.on("end", function() {
    console.log("end");
});

csvStream.on("error", function(err) {
    console.log(err);
    stream.destroy;
});
```

**Expected behavior**
Stream is not modified by parser. Calling parent is required to destroy stream upon parse failure.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: CentOS
 - OS Version: 7
 - Node Version: 12.16.3

**Additional context**

**Comment by foxmicha** [comment]
Here is the error received from my application: 
![image](https://user-images.githubusercontent.com/29128205/85149074-e0e3a600-b21e-11ea-8da3-037d7600cc5a.png)

**Comment by doug-martin** [comment]
Published `v4.3.1` to npm. Thanks again for the PR!
